ipa-core/src/protocol/basics/check_zero.rs

```diff
@@ -124,4 +187,122 @@ mod tests {
 
         Ok(())
     }
+
+    pub struct NotifyOnceCell<T: Clone + Send + Sync> {
```

Comment:
how is it different from https://docs.rs/tokio/latest/tokio/sync/struct.OnceCell.html? Maybe you could elaborate that in the documentation to this struct